crates/rust-analyzer/src/flycheck.rs

```diff
@@ -118,14 +118,14 @@ impl FlycheckHandle {
     }
 
     /// Schedule a re-start of the cargo check worker to do a workspace wide check.
-    pub(crate) fn restart_workspace(&self, saved_file: Option<AbsPathBuf>) {
-        self.sender.send(StateChange::Restart { package: None, saved_file }).unwrap();
+    pub(crate) fn restart_workspace(&self, saved_file: Option<AbsPathBuf>, target: Option<String>) {
```

Comment:
Workspace with target binary isn't a good fit (see comment below). So only allow passing target binary in `restart_for_package`.